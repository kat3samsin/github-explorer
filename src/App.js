import React, { Component } from 'react';
import Explorer from './Explorer';
import Projects from './Projects'

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className='title'>GitHub Project Explorer</h1>
        <Explorer />
        <Projects />
      </div>
    );
  }
}

export default App;
