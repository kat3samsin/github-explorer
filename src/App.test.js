import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', ()=> {
  it('renders without crashing', () => {
    let component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
})
