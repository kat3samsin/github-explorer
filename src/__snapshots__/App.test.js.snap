// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          href="https://github.com"
        >
          <img
            alt="GitHub"
            height="200px"
            src="https://image.flaticon.com/icons/svg/25/25231.svg"
            width="200px"
          />
        </a>,
        <h1
          className="title"
        >
          GitHub Repository Explorer
        </h1>,
        <UNDEFINED />,
        <UNDEFINED />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="GitHub"
            height="200px"
            src="https://image.flaticon.com/icons/svg/25/25231.svg"
            width="200px"
          />,
          "href": "https://github.com",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "GitHub",
            "height": "200px",
            "src": "https://image.flaticon.com/icons/svg/25/25231.svg",
            "width": "200px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "GitHub Repository Explorer",
          "className": "title",
        },
        "ref": null,
        "rendered": "GitHub Repository Explorer",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Explorer)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Projects)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            href="https://github.com"
          >
            <img
              alt="GitHub"
              height="200px"
              src="https://image.flaticon.com/icons/svg/25/25231.svg"
              width="200px"
            />
          </a>,
          <h1
            className="title"
          >
            GitHub Repository Explorer
          </h1>,
          <UNDEFINED />,
          <UNDEFINED />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="GitHub"
              height="200px"
              src="https://image.flaticon.com/icons/svg/25/25231.svg"
              width="200px"
            />,
            "href": "https://github.com",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "GitHub",
              "height": "200px",
              "src": "https://image.flaticon.com/icons/svg/25/25231.svg",
              "width": "200px",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GitHub Repository Explorer",
            "className": "title",
          },
          "ref": null,
          "rendered": "GitHub Repository Explorer",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Explorer)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Projects)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
